druid/src/widget/spinner.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An animated spinner widget.
+
+use std::f64::consts::PI;
+
+use druid::kurbo::Line;
+use druid::widget::prelude::*;
+use druid::{theme, Color, Data, KeyOrValue, Point, Vec2};
+
+/// An animated spinner widget for showing a loading state.
+///
+/// To customize the spinner's size, you can place it inside a [`SizedBox`]
+/// that has a fixed width and height.
+///
+/// [`SizedBox`]: struct.SizedBox.html
+pub struct Spinner {
+    t: f64,
+    color: KeyOrValue<Color>,
+}
+
+impl Spinner {
+    /// Create a spinner widget
+    pub fn new() -> Spinner {
+        Spinner {
+            t: 0.0,
+            color: theme::LABEL_COLOR.into(),
+        }
+    }
+
+    /// Builder-style method for setting the spinner's color.
+    ///
+    /// The argument can be either a `Color` or a [`Key<Color>`].
+    ///
+    /// [`Key<Color>`]: ../struct.Key.html
+    pub fn with_color(mut self, color: impl Into<KeyOrValue<Color>>) -> Self {
+        self.color = color.into();
+        self
+    }
+
+    /// Set the spinner's color.
+    ///
+    /// The argument can be either a `Color` or a [`Key<Color>`].
+    ///
+    /// [`Key<Color>`]: ../struct.Key.html
+    pub fn set_color(&mut self, color: impl Into<KeyOrValue<Color>>) {
```

Comment:
These two color methods should mention in their docs that the alpha channel is ignored.

Reply:
Ah that was an oversight. Fixed it to use the color's alpha.